guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/binary-agents/index.md

```diff
@@ -147,6 +147,29 @@ Make sure that each time you transcode a character from binary to decimal, you r
 
 *   <a href='http://forum.freecodecamp.com/t/javascript-array-prototype-map/14294' target='_blank' rel='nofollow'>Array.prototype.map</a>
 
+## ![:rotating_light:](https://forum.freecodecamp.com/images/emoji/emoji_one/rotating_light.png?v=3 ":rotating_light:") Advanced Code Solution Alternative:
+```javascript
+function binaryAgent(str) {
+  const re = /\d{8}\s*/g;
+  return str.replace(re, bin => String.fromCharCode(parseInt(bin, 2)));
+}
+
+// test here
+binaryAgent("01000001 01110010 01100101 01101110 00100111 01110100 00100000 01100010 01101111 01101110 01100110 01101001 01110010 01100101 01110011 00100000 01100110 01110101 01101110 00100001 00111111");
+```
+![:rocket:](https://forum.freecodecamp.com/images/emoji/emoji_one/rocket.png?v=3 ":rocket:") <a href='https://repl.it/repls/PointlessUltimateAddition' target='_blank' rel='nofollow'>Run Code</a>
```

Comment:
```suggestion
```
Hi, thank you for your PR but now we are no longer accepting links or other resources (images, etc.) which are not under our control, i.e. which a user or service might remove and is not under our control.

Please remove this line